commit 6257af2d05d460a0dd3e2a8268dd813edb980d81
Author: Andreas Schwab <schwab@redhat.com>
Date:   Thu Dec 1 13:27:53 2011 +0100

    Truncate time values in Linux futimes when falling back to utime

diff --git a/ChangeLog b/ChangeLog
index 2bf0a0a..dddb628 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -7,6 +7,12 @@
 	* misc/bits/select2.h (__FD_ELT): Mark as extension.  Add
 	parenthesis.
 
+2011-12-01  Andreas Schwab  <schwab@redhat.com>
+
+	* sysdeps/unix/sysv/linux/futimes.c: Truncate time values when
+	falling back to utime.
+
+
 2011-11-29  Andreas Schwab  <schwab@redhat.com>
 
 	* locale/weight.h (findidx): Add parameter len.
diff --git a/sysdeps/unix/sysv/linux/futimes.c b/sysdeps/unix/sysv/linux/futimes.c
index 272b83e..cd3cce6 100644
--- a/sysdeps/unix/sysv/linux/futimes.c
+++ b/sysdeps/unix/sysv/linux/futimes.c
@@ -1,5 +1,5 @@
 /* futimes -- change access and modification times of open file.  Linux version.
-   Copyright (C) 2002,2003,2005,2006,2007 Free Software Foundation, Inc.
+   Copyright (C) 2002,2003,2005,2006,2007,2011 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -95,8 +95,8 @@ __futimes (int fd, const struct timeval tvp[2])
       if (tvp != NULL)
 	{
 	  times = &buf;
-	  buf.actime = tvp[0].tv_sec + (tvp[0].tv_usec + 500000) / 1000000;
-	  buf.modtime = tvp[1].tv_sec + (tvp[1].tv_usec + 500000) / 1000000;
+	  buf.actime = tvp[0].tv_sec;
+	  buf.modtime = tvp[1].tv_sec;
 	}
       else
 	times = NULL;
